### PR TITLE
log ViewAndMutationMeta to trace_structured

### PR DESCRIPTION
**Pull request by bdhirsh**
I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


@torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131481
* #133044
* #137921
* __->__ #133784

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

File: torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py [hunk 1]
```diff
@@ -210,11 +210,21 @@ def _map_assigned_buffer_to_proxy(_mod, name, buffer):
...
+                    ),
+                ]
+            )
+
         trace_structured(
             "aot_forward_graph",
-            payload_fn=lambda: fw_module.print_readable(
-                print_output=False, include_stride=True, include_device=True
-            ),
+            payload_fn=lambda: fw_metadata_str,
```
Comment (ezyang) [inline torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:227]: This doesn't do what you want, payload_fn=fw_metadata_str

Have you inspected the output tlparse yet?

File: torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py [hunk 2]
```diff
@@ -210,11 +210,21 @@ def _map_assigned_buffer_to_proxy(_mod, name, buffer):
...
+        def fw_metadata_str():
+            return "\n\n".join(
+                [
+                    f"fw_metadata: {str(fw_metadata)}",
+                    f"maybe_subclass_meta: {str(maybe_subclass_meta)}",
+                    fw_module.print_readable(
+                        print_output=False, include_stride=True, include_device=True
+                    ),
+                ]
+            )
```
Comment (ezyang) [inline torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:223]: Why put these together? Just output them as separate log entries. If you don't want to have to modify tlparse cli, dump the fw metadata as "artifact"

File: torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py (hunk 1 above)
Comment (ezyang) [reply torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:227]: Oh it looks like you didn't exercise this one

**Comment by ezyang** [comment]
nit: I think the dumped structures would be substantially easier to understand if they were indented. There are two ways you could do this: dump as JSON, or use torchgen's dump dataclass with indentation utility

File: torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py (hunk 1 above)
Comment (bdhirsh) [reply torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:227]: whoops... thank you

File: torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py (hunk 2 above)
Comment (bdhirsh) [reply torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:231]: you're completely right - I didn't realize that dumping using "artifact" allows us to add new artifacts without needing to change the tlparse cli. updated

**Comment by bdhirsh** [comment]
Just updated to use `dataclass_repr()` for formatting and it is much nicer https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp2lEs4q/0_0_1/aot_forward_graph_fw_metadata_2.txt

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**